.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: Build
+on:
+  push:
+    branches: [main]
+  pull_request:
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@19b5e6071f70a82d3c5741d7b3f98a71d192eb20
+        with:
+          environments: build
+      - name: Build project
+        run: pixi run -e build build-wheel
+      - name: Upload package
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifact
+          path: dist/*
+
+  release:
+    name: Publish package
+    if: github.event_name == 'push' && github.ref_name == 'main'
```

Comment:
This shouldn't be on _every_ push on main, i don't want to pollute pypi with random releases... (testpypi would be okay)